core/logger/zap.go

```diff
@@ -5,27 +5,99 @@ import (
 	"fmt"
 	"io"
 	"os"
+	"path/filepath"
 
 	"go.uber.org/zap"
 	"go.uber.org/zap/zapcore"
+	"gopkg.in/natefinch/lumberjack.v2"
 )
 
 var _ Logger = &zapLogger{}
 
+// ZapLoggerConfig defines the struct that serves as config when spinning up a the zap logger
+type ZapLoggerConfig struct {
+	zap.Config
+	local Config
+	sinks []zapcore.WriteSyncer
+}
+
 type zapLogger struct {
 	*zap.SugaredLogger
-	config     zap.Config
+	config     ZapLoggerConfig
 	name       string
 	fields     []interface{}
 	callerSkip int
 }
 
-func newZapLogger(cfg zap.Config) (Logger, error) {
-	zl, err := cfg.Build()
-	if err != nil {
-		return nil, err
+func newZapLogger(cfg ZapLoggerConfig) (Logger, error) {
+	sinksCores := []zapcore.Core{
+		newConsoleCore(cfg),
 	}
-	return &zapLogger{config: cfg, SugaredLogger: zl.Sugar()}, nil
+
+	if cfg.local.ToDisk {
+		sinksCores = append(sinksCores, newDiskCore(cfg.local))
+	}
+
+	for _, sink := range cfg.sinks {
+		sinksCores = append(
+			sinksCores,
+			zapcore.NewCore(
+				zapcore.NewJSONEncoder(makeEncoderConfig(cfg.local)),
+				sink,
+				zap.LevelEnablerFunc(cfg.Level.Enabled),
+			),
+		)
+	}
+
+	core := zapcore.NewTee(sinksCores...)
+
+	return &zapLogger{config: cfg, SugaredLogger: zap.New(core).Sugar()}, nil
+}
+
+func newDiskCore(cfg Config) zapcore.Core {
+	var (
+		encoder = zapcore.NewJSONEncoder(makeEncoderConfig(cfg))
+		sink    = zapcore.AddSync(&lumberjack.Logger{
+			Filename:   filepath.ToSlash(filepath.Join(cfg.Dir, "log.jsonl")),
```

Comment:
Can we have the disk logs be normal uncolored terminal output, not json?

Reply:
I changed the file format for it, but `zap` will still log the data in JSON format regarding of the `encoder` you choose. any thoughts @jmank88 ?

The `ConsoleEncoder` they have states the following:

```
// NewConsoleEncoder creates an encoder whose output is designed for human -
// rather than machine - consumption. It serializes the core log entry data
// (message, level, timestamp, etc.) in a plain-text format and leaves the
// structured context as JSON.
```

Reply:
I have validated that the logs are being now stored as terminal outputs instead. Apparently my executable was not updated/re-built. We are fine here.